### DISCUSSION TITLE
How to Make API Requests from React Native / Expo

### DISCUSSION THREAD

**Discussion by mayone-du** [honojs/hono#2844] [Q&A] [answered]
I'm developing an app using React Native / Expo. To make fetch requests to an API running on a local PC from Expo Go, you need to make requests in the format of private_ip:8787 rather than localhost:8787. However, in the case of Hono, this approach resulted in errors and didn't work properly. Could you please tell me the cause of this and how to resolve it?

**Comment by EdamAme-x** [comment]
Please show me details of error

**Reply by EdamAme-x** [reply]
I'm sorry if this is not the case, but isn't it a configuration error on the network side?
Is JavaScript Runtime throwing an error?

**Reply by mayone-du** [reply]
Thank you for your response!

Here's what I'm trying:
Running `npm run dev` (wrangler dev src/index.ts) with Hono.

The local server starts at `http://localhost:8787`, and I can access it from the browser without any issues.

However, when I replace the localhost part with the private IP connected on my PC, the server doesn't respond (`192.168.40.198:8787`).

As a test, I've launched other projects (NestJS or Hasura), and accessing `192.168.40.198:4000` or `192.168.40.198:8080/console` returns responses successfully. Therefore, I believe this might be related to Hono or Miniflare. What do you think?

**Reply by mayone-du** [reply]
（When accessing 192.168.40.198:8787, no logs are flowing in Hono, so it seems like the requests are not reaching it in the first place.）

**Reply by EdamAme-x** [reply]
Thank for the reply.
Could you please show me the minimum code? (You may omit the logic part.)

**Reply by mayone-du** [reply]
I've created a repository to share, so please take a look here:
https://github.com/mayone-du/expo-hono-monorepo-example

**Reply by EdamAme-x** [reply]
thanks for the reply. I'll check it later.

**Reply by EdamAme-x** [reply]
I have two questions.
1. can you access "private_ip:8787" on the browser?
2. do you opening a port at 8787?

**Reply by EdamAme-x** [reply]
If it is a problem on the Hono-side, then it is a problem with the Hono-client, but it does not appear to be a problem.

**Reply by mayone-du** [reply]
> 1. can you access "private_ip:8787" on the browser?

no

> 2. do you opening a port at 8787?

yes

**Reply by EdamAme-x** [reply]
thanks.
hmm... 
Is “wrangler” the cause...?
Please create a handler that accepts all paths (same as "*") and try accessing it again. If you still can't do it, it may be a problem on the wrangler side.

**Reply by EdamAme-x** [reply]
It's not a network problem, such as firewalls being blocked, is it?

**Reply by mayone-du** [reply]
> It's not a network problem, such as firewalls being blocked, is it?

I believe it's not a network issue since the firewall is turned off.

**Reply by EdamAme-x** [reply]
hmm... I don't know...

**Reply by mayone-du** [reply, marked as answer]
Thank you for thinking together with me! We resolved it by adding the ip option to the wrangler command 🎉

npm scripts↓

```diff
- "dev": "wrangler dev src/index.ts --persist-to=./localdb",
+ "dev": "wrangler dev src/index.ts --persist-to=./localdb --ip=192.168.40.198",
```

**Reply by EdamAme-x** [reply]
good
